openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -256,23 +267,84 @@ private void setTestSuiteSummary(TestCase testCase, Long timestamp, TestCaseStat
     for (TestSuite testSuite : testSuites) {
       testSuite.setSummary(null); // we don't want to store the summary in the database
       List<ResultSummary> resultSummaries = listOrEmpty(testSuite.getTestCaseResultSummary());
-      if (resultSummaries.isEmpty()) {
-        resultSummaries.add(resultSummary);
-      } else {
-        // We'll remove the existing summary for this test case and add the new one
+      if ((isDeleted) && (resultSummaries.isEmpty())) {
```

Comment:
this should be OR condition? we will not add if the testCase is deleted or there are not results?

Reply:
This handles cases where you would try to delete a testCaseResult and the resultSummaries (which holds the state) is already empty. In this scenario there is nothing to do. Let me know if that makes sense.